client/src/components/StreamField/index.js

```diff
@@ -0,0 +1 @@
+import '../../../../node_modules/react-streamfield/dist/react-streamfield.js';
```

Comment:
Judging by how this code is included, I understand that the compiled `dist/react-streamfield.js` file includes all of the library's dependencies directly bundled? If so, this means that React + Redux and all their related libraries will be imported twice, among other issues (lack of package management for transitive dependencies). It would be better if:

- Libraries already used by Wagtail were marked as peer dependencies, so we are guaranteed things like React & Redux are only included once, with the same version everywhere.
- Other libraries should be declared as `dependencies` in the package.json, so Wagtail's client-side build can leverage npm to manage their versions (de-duping, & transitive dependency updates without having to re-release `react-streamfield` every time.

I would also expect some or all of the code in https://github.com/noripyt/react-streamfield/blob/c8a42f7700724e603de494d2432e3e966c5c6ca6/src/index.js to be defined in Wagtail instead of the library. It's more common for libraries based on React components to expose the components for integration, or at least an initialisation function, so the dependencies of the library are more clearly defined. Imports should be free of side-effects.

Reply:
I fully agree with you, this is the ideal solution.
I originally wanted to do exactly this, as you can see on the initial commit: https://github.com/wagtail/wagtail/pull/4942/commits/884e363da094a74d2623ce4c1bcec7c317083643

Unfortunately, I couldn’t manage to update all other projects to Babel 7, I kept facing conflicts, even after twenty hours spent on it… So I decided to make it a fully external dependency instead.

I also agree that `react-streamfield` itself must only be made of components and not have this auto-loaded logic in its `index.js`. But that’s for a scenario where the components API would be stabilized and documented, which I do not plan to do in a near future. It’s out of scope of this work, as discussed in https://github.com/noripyt/react-streamfield/issues/3. So for now, the API is just pure JSON. I’m perfectly aware that it’s not the ideal React approach, but it’s still a perfectly valid React use case.

Reply:
Thank you for the PR you did on react-streamfield for that :)